modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java

```diff
@@ -0,0 +1,294 @@
+
+        //Add the application throttling policy
+        ApiResponse<ApplicationThrottlePolicyDTO> addedPolicy2 = restAPIAdmin.addApplicationThrottlingPolicy(
+                applicationThrottlePolicyDTO2);
+
+        //Assert the status code and policy ID
+        Assert.assertEquals(addedPolicy2.getStatusCode(), HttpStatus.SC_CREATED);
+        applicationThrottlePolicyDTO2 = addedPolicy2.getData();
+        Assert.assertNotNull(applicationThrottlePolicyDTO2.getPolicyId(), "The policy ID cannot be null or empty");
+        applicationThrottlePolicyDTO2.setIsDeployed(true);
+
+        // create api
+        backendEP = gatewayUrlsWrk.getWebAppURLNhttp() + "response/";
+        String url = backendEP;
+        APIRequest apiRequest = new APIRequest(APIName, APIContext, new URL(url), new URL(url));
+        apiRequest.setVersion(APIVersion);
+        apiRequest.setSandbox(url);
+        apiRequest.setProvider(user.getUserName());
+        apiRequest.setTiersCollection(Constants.TIERS_UNLIMITED);
+        List<APIOperationsDTO> operations = new ArrayList<>();
+        APIOperationsDTO apiOperationsDTO = new APIOperationsDTO();
+        apiOperationsDTO.setVerb("POST");
+        apiOperationsDTO.setTarget("/*");
+        apiOperationsDTO.setAuthType("Application & Application User");
+        apiOperationsDTO.setThrottlingPolicy("Unlimited");
+        operations.add(apiOperationsDTO);
+        apiRequest.setOperationsDTOS(operations);
+
+        HttpResponse serviceResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId = serviceResponse.getData();
+        // Create Revision and Deploy to Gateway
+        createAPIRevisionAndDeployUsingRest(apiId, restAPIPublisher);
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, Constants.PUBLISHED);
+        waitForAPIDeploymentSync(user.getUserName(), APIName, APIVersion, APIMIntegrationConstants.IS_API_EXISTS);
+
+        // check backend
+        Map<String, String> requestHeaders = new HashMap<String, String>();
+        HttpResponse response = HttpRequestUtil.doGet(backendEP, requestHeaders);
+        Assert.assertEquals(response.getResponseCode(), 200, "Backend (dummy_api.xml) is not up and running");
+
+        // Create an Application for request count limit
+        requestCountApplicationDTO = restAPIStore.addApplication(applicationName1, policyName1, "",
+                "this-is-test-application-for-request-count-limit");
+
+        // Create an Application for bandwidth limit
+        bandwidthApplicationDTO = restAPIStore.addApplication(applicationName2, policyName2, "",
+                "this-is-test-application-for-bandwidth-limit");
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test reset application throttling policy with request count limit")
+    public void testResetPolicyWithRequestCountLimit() throws Exception {
+        //subscribe to API
+        SubscriptionDTO subscriptionDTO = restAPIStore.subscribeToAPI(apiId,
```

Comment:
Move subscription code to setEnvironment method

Reply:
Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753